overlord/snapstate/component.go

```diff
@@ -170,9 +170,11 @@ func doInstallComponent(st *state.State, snapst *SnapState, compSetup *Component
 		addTask(kmodSetup)
 	}
 
+	changingSnapRev := !snapst.Current.Unset() && snapst.Current != snapSi.Revision
```

Comment:
Can the current revision of a snap be unset? Maybe I am missing something, but would not this happen only if it is not installed yet? But then `SnapState.IsInstalled()` should be used.

Reply:
Good call, yes this should just be IsInstalled